api/src/repositories/summary-repository.ts

```diff
@@ -428,6 +428,20 @@ export class SummaryRepository extends BaseRepository {
       throw failedToFindValidationRulesError;
     }
 
+    // check if the species in the survey and the species on the template match
+    if (species) {
+      if (!response.rows.some((row) => species.includes(Number(row.wldtaxonomic_units_id)))) {
```

Comment:
Could be a good idea to write a couple tests covering these two branches

Reply:
good call, I've added some test around this code